lib/spack/spack/cray_manifest.py

```diff
@@ -74,13 +78,13 @@ def spec_from_entry(entry):
 
     compiler_str = ""
     if "compiler" in entry:
-        compiler_format = "%{name}@{version}"
+        compiler_format = "%{name}@={version}"
         compiler_str = compiler_format.format(
             name=translated_compiler_name(entry["compiler"]["name"]),
             version=entry["compiler"]["version"],
         )
 
-    spec_format = "{name}@{version} {compiler} {arch}"
+    spec_format = "{name}@={version} {compiler} {arch}"
```

Comment:
I think I like the default range, but this does seem a little tricky.  Basically any place we read things in we have to be careful to interpret the version concretely.

This is actually not unlike how Specs were originally treated in Spack -- Specs read from files (like the database) had to be interpreted as implicitly concrete before we added the `concrete` attribute to them.  It defaults to `True` so you don't see it in the DB, but if you output an abstract spec it'll have `concrete=False` in it.

I am not sure how to avoid the need to be careful about this so I guess I'll just leave this as a comment.

Reply:
Agreed, also when you copy a literal string from `spack spec` and feed it back into `spack spec` again, it might concretize to a "more specific" version. Alternatively `str(StandardVersion)` should print as `=<version>`, but I didn't go for that because it felt like a big change.